main.cpp

```diff
@@ -108,12 +109,27 @@ void uthread_create(void *(start_routine)(void *), void* arg){
     auto sp = (address_t)stack + STACK_SIZE - 10 * sizeof(void*);
     auto pc = (address_t)thread_wrapper;
 
+    // Place the argument and the function pointer on the stack
+    *sp = (address_t)arg;
+    sp--;
+    *sp = (address_t)start_routine;
+    sp--;
+
+    // Place the wrapper function on the stack so when the scheduler switches
+    // to this thread, the wrapper function is popped off the stack
+    *sp = (address_t)thread_wrapper;
+    sp--;
+
```

Comment:
When we switch to this thread, the thread_wrapper function will be popped of the stack to be the new pc

Reply:
Could you test with `g++ main.cpp && ./a.out`? For me that doesn't work.

Reply:
Alright I'll work on it

Reply:
On second thought, I'm guessing maybe we can just ignore pushing the argument and the and the function on the stack because putting them in the tcb will work fine. I'll just double check with the TA

Reply:
I just clarified with Francis that we don't need to use the stack. I'm not going to bother trying to figure out how to do it the stack way right now. Let's just leave it the way it was and forget these changes.